### ISSUE TITLE
No connection possible since adapter update to 3.0.6 or 4.0

### ISSUE THREAD

**Issue by mickym2** [ioBroker/ioBroker.mqtt#291]
After the adapter became part of the stable repository I updated the mqtt adapter but couldnt connect to my mosquitto mqtt- Broker. 

So mqtt Adapter is used in Client  mode.

After the update was performed successfully I entered my password again. The connection test within the adapter itself was successful, but the connection failed.  

`mqtt.1 2022-05-12 23:52:50.539	error	Client error:Error: Connection refused: Not authorized
`

The mosquitto log looks as if the adapter connects without client-ID 

`
1652392619: New connection from 127.0.0.1:39240 on port 1883.

1652392619: Client <unknown> disconnected, not authorised.

1652392625: New connection from 127.0.0.1:39242 on port 1883.

1652392625: Client <unknown> disconnected, not authorised.
`

**Comment by Apollon77** [comment]
Please provide a Debug log from Adapterstart

**Comment by mickym2** [comment]
So it seems - that even the inplace upgrade sent a rc=0 - that the first time the upgrade runs not smoothly. I stopped all mqtt adapters, used a new userid and password and now it works. So i will close this for now - but have a final question. I can't see anymore the mask of the published states in version 3.0.6 - so must I upgrade to version 4.x?

![image](https://user-images.githubusercontent.com/58858526/168430583-a63e3324-812c-4b5b-b156-c353640efacd.png)

**Comment by mickym2** [comment]
First installation or user credentials were not correctly taken over - even after new input of the password. I did a second time and now 3.0.6 is working - but I cant see in one instance the mask which topics are published by this instance.

**Comment by Apollon77** [comment]
> but I cant see in one instance the mask which topics are published by this instance.

This is only displayed when "server" in the olf version and new version changes oublish beaviour completely

**Comment by mickym2** [comment]
But this was not the case in version 2.7.4 - i could use a separate instance to publish iobroker states into a mqtt broker. I created another issue. I understand that it was not tolerable that the mqtt adapter could write into other adapter spaces - but I do not understand why only publishing is no longer possible. I created an request here and opened a thread. 

And the mask to add additional topics was definetely available in version 2.7.x and earlier regardless if server or client. I used this for years now.

https://github.com/ioBroker/ioBroker.mqtt/issues/293
https://forum.iobroker.net/topic/54903/mqtt-kein-publish-von-iob-states-ausserhalb-mqtt-scopes
